sql/core/src/main/scala/org/apache/spark/sql/execution/command/tables.scala

```diff
@@ -791,11 +791,22 @@ case class ShowCreateTableCommand(table: TableIdentifier) extends RunnableComman
     }
   }
 
+  // These table properties should not be included in the output statement of SHOW CREATE TABLE
+  val excludedTableProperties = Set(
+    // The following are hive-generated statistics fields
+    "COLUMN_STATS_ACCURATE",
+    "numFiles",
+    "numPartitions",
+    "numRows",
+    "rawDataSize",
+    "totalSize"
+  )
+
   private def showHiveTableProperties(metadata: CatalogTable, builder: StringBuilder): Unit = {
     if (metadata.properties.nonEmpty) {
       val filteredProps = metadata.properties.filterNot {
-        // Skips "EXTERNAL" property for external tables
-        case (key, _) => key == "EXTERNAL" && metadata.tableType == EXTERNAL
+        // Skips all the stats info (See the JIRA: HIVE-13792)
```

Comment:
Not sure whether I got your point. Are you saying `getTableOption` in `HiveExternalCatalog`?

Initially, I had the same design like you. Later, I realized we still use/display them in the other DDL statements, for example, `DESCRIBE EXTENDED TABLE`. 

Let me know if my understanding is wrong. Thanks!